subprojects/platform-native/src/main/java/org/gradle/nativeplatform/internal/resolve/SourceSetNativeDependencyResolver.java

```diff
@@ -59,21 +59,17 @@ private NativeDependencySet createNativeDependencySet(LanguageSourceSet sourceSe
     private static class EmptyNativeDependencySet implements NativeDependencySet {
         @Override
         public FileCollection getIncludeRoots() {
-            return empty();
+            return ImmutableFileCollection.of();
         }
 
         @Override
         public FileCollection getLinkFiles() {
-            return empty();
+            return ImmutableFileCollection.of();
```

Comment:
Should we add an empty-parameter factory method for this guy, I wonder? It would mean one less allocation of an empty `Object[]`. Not a big deal... We could also name it `empty()`, but I'm also buying into keeping it named `of()` for the reasons Guava uses. (Which is that when you add/remove elements, you shouldn't need to change the method name.)